File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<int, char>mp; int l=0,res=0,r=0;
        for(r;r<s.size();r++){
            if(mp.find(s[r]) != mp.end()) l=max(mp[s[r]]+1,l);
            mp[s[r]]=r;
            res=max(res,r-l+1);
        }
        return res;
    }
};

File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        l,r=0, len(height)-1
        res=0
        lm,rm=height[l], height[r]
        while l<r:
            if lm<=rm:
                l+=1
                lm=max(lm,height[l])
                res+=lm-height[l]
            else:
                r-=1
                rm=max(rm,height[r])
                res+=rm-height[r]
        return res